lm_eval/evaluator.py

```diff
@@ -481,33 +465,96 @@ def evaluate(
                     results[task_name][metric + "_stderr" + "," + key] = stderr(items)
 
         if bool(results):
-            for task_or_group in results.keys():
-                for metric in results[task_or_group].keys():
-                    if type(results[task_or_group][metric]) == list:
-                        if "(sample agg)" in metric:
-                            results[task_or_group][metric] = sample_agg_fn[
-                                task_or_group
-                            ][metric](results[task_or_group][metric])
-                        else:
-                            results[task_or_group][metric] = np.average(
-                                results[task_or_group][metric]
-                            )
-                        versions[task_or_group] = "N/A"
 
-        for task_name, task in task_dict.items():
-            if type(task) == tuple:
-                group_name, task = task
+            for group, task_list in reversed(task_hierarchy.items()):
```

Comment:
We should split this logic out into one or a few sub-methods, this gets to be fairly unreadable. I could handle this since you'll be away this next week